### PR TITLE
Gherkin Scripts for epic ODC-4388

### PR DESCRIPTION
**Pull request by gajanan-more**
Fixes: https://issues.redhat.com/browse/ODC-4823

Problem: Gherkin Scripts for epic [ODC-4388](https://issues.redhat.com/browse/ODC-4388)

Solution: This PR have the gherkin scripts for all the scenarios related to [ODC-4388](https://issues.redhat.com/browse/ODC-4388).

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/features/operators.feature [hunk 1]
```diff
@@ -105,3 +105,27 @@ Scenario: Uninstall the Pipeline Operator from Operator Hub page : P-013-TC01, P
...
+
+
+@regression, @smoke
+Scenario: Create Web Terminal instance
+   Given user is at Installed Operator page with the header name "Installed Operators"
+   And user has installed Web Terminal Operator
+   When user clicks on Web Terminal operator
+   And user clicks on All Instances tab
+   And user clicks on Create New button
+   And user clicks on DevWorkspace
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/operators.feature:128]:
```suggestion
   And user clicks on Workspace
```

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature [hunk 2]
```diff
@@ -0,0 +1,28 @@
+Feature: Web Terminal for Developer user
+    As a user with admin rights, I should be able to use web terminal
+
+
+Background:
+    Given user has logged in as admin user
+    And user is at developer perspecitve
+    And user has installed Web Terminal operator
+    And user has created Web Terminal instance
+    And user has created namespace "aut-terminal-admin"
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature:10]:
```suggestion
    And user has selected namespace "aut-terminal-admin"
```

Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature:10]: I think we should remove this line, As In first scenario we are using project creation

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-devuser.feature [hunk 3]
```diff
@@ -0,0 +1,28 @@
+Feature: Web Terminal for Developer user
+    As a developer user, I should be able to use web terminal
+
+
+Background:
+    Given user has logged in as basic user
+    And user is at developer perspecitve
+    And user has installed Web Terminal operator
+    And user has created Web Terminal instance
+    And user has created namespace "aut-terminal-dev"
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-devuser.feature:10]: Remove this line

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+Background:
+    Given user is at developer perspecitve
+    And user has installed Web Terminal operator
+    And user has created Web Terminal instance
+    And user has created namespace "aut-terminal-basic"
+    And user is at the Topolgy page
+
+
+@regression, @smoke
+Scenario: Web Terminal icon Masthead
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature:14]: Remove this scenario, IMO this is not required

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+
+@regression, @smoke
+Scenario: Web Terminal window
+    Given user is at topology page
+    When user clicks on the Web Terminal icon on the Masthead
+    Then user will see the terminal window
+
+
+@regression, @smoke
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature:26]: Do you think this is required for smoke test suite

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature (hunk 2 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature:10]: There are two different scenarios, one is selecting already created namespace and the other one is creating namespace in the terminal window itself. That's why I added that line.

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-devuser.feature (hunk 3 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-devuser.feature:10]: Same as above

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature (hunk 4 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature:14]: Removed

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature (hunk 5 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature:26]: Kept in regression only

File: frontend/packages/dev-console/integration-tests/features/operators.feature (hunk 1 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/operators.feature:128]: Updated

File: frontend/packages/dev-console/integration-tests/features/operators.feature [hunk 6]
```diff
@@ -105,3 +105,27 @@ Scenario: Uninstall the Pipeline Operator from Operator Hub page : P-013-TC01, P
...
+Scenario: Create Web Terminal instance
+   Given user is at Installed Operator page with the header name "Installed Operators"
+   And user has installed Web Terminal Operator
+   When user clicks on Web Terminal operator
+   And user clicks on All Instances tab
+   And user clicks on Create New button
+   And user clicks on Workspace
+   And user clicks on Create button
+   Then user will be redirected to Workspace tab
+   And user will see "web-terminal" instance created
```
Comment (christianvogt) [inline frontend/packages/dev-console/integration-tests/features/operators.feature:131]: Useful I suppose but not related to how we integrate web terminal into console. As a user of the integrated terminal, there is no expectation to create a workspace in this manner.

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature (hunk 2 above)
Comment (christianvogt) [reply frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature:10]: Looks to me like the scenario could simply use _any_ namespace to begin with. As the starting namespace is irrelevant if we are creating a new namespace.

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    And user is at the Topolgy page
+
+
+@regression, @smoke
+Scenario: Create new project and use Web Terminal
+    Given user is at topology page
```
Comment (christianvogt) [inline frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature:16]: Does the scenario need to start on a particular page?
The starting page here is irrelevant to the test.

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+    When user clicks on the Web Terminal icon on the Masthead
+    And user selects Create Project from Project drop down menu
+    And user enters project name "aut-terminal-adminuser"
+    Then user will see the terminal window for namespace "aut-terminal-adminuser"
```
Comment (christianvogt) [inline frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature:20]: Is there normally a step in between here that requires the user to submit the form?

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+    As a user with admin rights, I should be able to use web terminal
+
+
+Background:
+    Given user has logged in as admin user
+    And user is at developer perspecitve
+    And user has installed Web Terminal operator
+    And user has created Web Terminal instance
+    And user has created namespace "aut-terminal-admin"
+    And user is at the Topolgy page
```
Comment (christianvogt) [inline frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature:11]: Why does the page matter?

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+    Given user is at topology page
+    When user clicks on the Web Terminal icon on the Masthead
+    Then user will see the terminal window
+
+
+@regression
+Scenario: Web Terminal in new tab
+    Given user is at topology page
+    When user clicks on the Web Terminal icon on the Masthead
+    And user clicks on Open Terminal in new tab button
```
Comment (christianvogt) [inline frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature:24]: should `Open Terminal in new tab` be quoted?

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature (hunk 2 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature:10]: Okay. I will remove this scenario.

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-devuser.feature (hunk 3 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-devuser.feature:10]: Sure

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature (hunk 10 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature:24]: In this statement, we will be clicking on a button present on the page, so I don't think we should quote anything.

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature (hunk 7 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature:16]: It's not mandatory but we have to start somewhere. That's why I have to add that statement.

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature (hunk 9 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature:11]: In the background, it doesn't matter. I will remove it.

File: frontend/packages/dev-console/integration-tests/features/operators.feature (hunk 6 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/operators.feature:131]: @christianvogt  I didn't get it. For now, we have to create the workspace to get the icon. Will there be change in the process and we don't gonna need it? Should I remove this?

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature (hunk 8 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature:20]: It's there. I will add a new statement where user submits the form.

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature [hunk 11]
```diff
@@ -0,0 +1,27 @@
+Feature: Web Terminal for Developer user
+    As a user with admin rights, I should be able to use web terminal
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature:2]: Feature and statement do not match. If it is for the user with admin rights then the feature should be for the Admin user

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature [hunk 12]
```diff
@@ -0,0 +1,24 @@
+Feature: Web Terminal
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature:1]: Web terminal in developer perspective

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+    Given user is at topology page
+    When user clicks on the Web Terminal icon on the Masthead
+    Then user will see the terminal window
+
+
+@regression
+Scenario: Web Terminal in new tab
+    Given user is at topology page
+    When user clicks on the Web Terminal icon on the Masthead
+    And user clicks on Open Terminal in new tab button
```
Comment (sanketpathak) [inline frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature:23]: I think one more step should be there- `And user sees the terminal window`

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature (hunk 11 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature:2]: Updated

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature (hunk 12 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature:1]: Web Terminal can be opened in any perspective. That's I have kept it like that.

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature (hunk 13 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature:23]: I haven't added a separate statement but updated one statement.

Comment (sanketpathak) [reply frontend/packages/dev-console/integration-tests/features/web-terminal/web-teminal-basic.feature]: For automating the scenario we need to wait for the Terminal window to open right?

File: frontend/packages/dev-console/integration-tests/features/operators.feature (hunk 6 above)
Comment (christianvogt) [reply frontend/packages/dev-console/integration-tests/features/operators.feature:131]: A console user isn't expected to manually create a Workspace. They will do so through the terminal UI by clicking the icon and filling out the form.

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature [hunk 14]
```diff
@@ -0,0 +1,27 @@
+Feature: Web Terminal for Admin user
+    As a user with admin rights, I should be able to use web terminal
+
+
+Background:
+    Given user has logged in as admin user
+    And user is at developer perspecitve
+    And user has installed Web Terminal operator
+    And user has created Web Terminal instance
```
Comment (christianvogt) [inline frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature:9]: What is a web terminal instance?

Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature:9]: It was referring to 'Create Web Terminal instance' scenario. But I have removed this scenario so I will remove that statement as well.

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-devuser.feature [hunk 15]
```diff
@@ -0,0 +1,26 @@
+Feature: Web Terminal for Developer user
+    As a developer user, I should be able to use web terminal
+
+
+Background:
+    Given user has logged in as basic user
+    And user is at developer perspecitve
+    And user has installed Web Terminal operator
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-devuser.feature:8]:
```suggestion
    And user has installed Web Terminal operator
    And user is at developer perspective
```

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature [hunk 16]
```diff
@@ -0,0 +1,26 @@
+Feature: Web Terminal for Admin user
+    As a user with admin rights, I should be able to use web terminal
+
+
+Background:
+    Given user has logged in as admin user
+    And user is at developer perspecitve
+    And user has installed Web Terminal operator
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature:8]:
```suggestion
    And user has installed Web Terminal operator
    And user is at developer perspective
```

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature (hunk 7 above)
Comment (makambalaji) [reply frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature:13]: Please remove topology page from given statement for all scenarios, as this web terminal is available on master head

**Review by makambalaji — COMMENTED** [review]
Please update as per the comments

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-devuser.feature (hunk 15 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-devuser.feature:8]: Updated

File: frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature (hunk 16 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/web-terminal/web-Terminal-adminuser.feature:8]: Updated

**Pull request merged**